### ISSUE TITLE
Setting value to Infinity prints NaNaNaNaNaNaNaNaN,00

### ISSUE THREAD

**Issue by thedjinn** [s-yadav/react-number-format#256]
Suppose I have a variable that I supply with an Infinity value:

```javascript
const value = 1 / 0;
```

When I provide the value to `NumberFormat` the text value of the HTML input tag is `NaNaNaNaNaNaNaNaN,00`.

I am using the following props:

```jsx
<NumberFormat
    thousandSeparator="."
    decimalSeparator=","
    decimalScale={2}
    fixedDecimalScale
    value={value}
/>
```

**Comment by ahmadsaeed90** [comment]
I got the same issue when displaying non number strings, the reason was "decimalScale" prop. The only solution that worked for me was adding condition to the decimalScale e.g
decimalScale={ isFinite(value)? 2: 0 }

Also if you still get error, make fixedDecimalScale conditional too.

Hope this helps.

**Comment by s-yadav** [comment]
Fixed this on v4.0.6.
